### ISSUE TITLE
night mode implementation? [ft][en] request

### ISSUE THREAD

**Issue by mstojanovic** [owncloud/android#2512]
are there any plans to introduce a night mode for the android app? that would be a great feature/enhancement imho. unfortunately gui design is not something im good at at all (i really dont know how to pair colors) but maybe i could help a bit in the background during the implementation

**Comment by michaelstingl** [comment]
@mstojanovic You mean a dark theme a user can select or something different? Could you elaborate? Do you know an example of an Android with this feature?

**Comment by mstojanovic** [comment]
@michaelstingl yes, i was thinking about a dark theme. i dont think that should be to hard to implement (to each xml in drawable add the definitions for the themes, change the res/layout files to have a dynamic value for the layout, add a switch for the user to choose). the idea is that a user can go to e.g. the settings of the app and change the theme (background color, font color, ...). many apps offer that feature, e.g. twitter has a night mode.

**Comment by michaelstingl** [comment]
You can already change colors if you compile your own client:
https://github.com/owncloud/android/blob/b549f879e3f6ab25d73721d59e33a690181035bc/owncloudApp/src/main/res/values/setup.xml#L35-L40
https://github.com/owncloud/android/blob/57fd78e326e0eb8187fe87d001cdb81eb4504188/owncloudApp/src/main/res/values/colors.xml#L27-L30

We're experimenting with more flexible theming options in the new iOS app:
https://github.com/owncloud/ios-app/issues/251
I could imagine bringing similar methods to the Android app as well.

At the moment the focus of the team is to rework the architecture to make the app more reliable and future proof:
https://github.com/owncloud/android/labels/Architecture

**Comment by mstojanovic** [comment]
i dont think you get my point... im aware that one can change the colors if he builds his own app. my idea is to give any user an option to choose a theme he wants to use (not to mention that darker themes can save battery life on AMOLED screens).

My problem with doing the changes is that i dont have the knowledge to make the gui visually appealing with color choices, but i do know how to do the technical part of implementing it. the goal of this issue/request is for the team members to think about it and maybe propose or point me into the direction of a official color pallet. or maybe give some community member an idea and she or he can come up with a color pallet that would work

**Comment by michaelstingl** [comment]
Here's the iOS example:

![2019-04-04 10 39 43](https://user-images.githubusercontent.com/214010/55542014-76e8ea80-56c6-11e9-9eda-fa4f1176f40e.png) | ![2019-04-04 10 39 53](https://user-images.githubusercontent.com/214010/55542015-76e8ea80-56c6-11e9-8588-ea399ee320e1.png) | ![2019-04-04 10 40 03](https://user-images.githubusercontent.com/214010/55542016-76e8ea80-56c6-11e9-8aac-8cdc59f452dd.png) | ![2019-04-04 10 40 21](https://user-images.githubusercontent.com/214010/55542017-77818100-56c6-11e9-9574-95081f01c902.png)
-- | -- | -- | --

You'll find some pointers in this file:
https://github.com/owncloud/ios-app/blob/master/ownCloud/Theming/ThemeCollection.swift

There's also an ongoing project to document the ownCloud design:
https://github.com/owncloud/owncloud-design-system

To sum it up:
I'd recommend to wait a few more weeks. Then there will be a new set of color parameters designed to work in the web UI, iOS app and Android app, and a documentation about the actual color values. Feel free to ping again to learn about the progress.

**Comment by mstojanovic** [comment]
great, ty! thats all the info i needed :)

**Comment by davigonz** [comment]
Android 10 brought the system-wide dark theme , we could refloat this, what do you think @michaelstingl ?

**Comment by michaelstingl** [comment]
@davigonz yeah, fits well in the epic https://github.com/owncloud/android/issues/2742

**Comment by DeepDiver1975** [comment]
🤖 Automated issue triage by Claude Code review agent.

This issue was opened before 2020 and has had no activity since then, so we are
closing it as outdated. If this is still relevant on a current ownCloud release,
please **reopen** (or open a fresh issue) with up-to-date details — thank you\!
